core/vm/evm.go

```diff
@@ -352,8 +352,21 @@ func (evm *EVM) StaticCall(caller ContractRef, addr common.Address, input []byte
 	return ret, contract.Gas, err
 }
 
+type codeAndHash struct {
+	code []byte
+	hash *common.Hash
```

Comment:
Same here. Generally our code considers `common.Hash{}` to be the nil/uninited hash. Do we really need the pointers?